Slim/Plugin/WiMP/ProtocolHandler.pm

```diff
@@ -269,29 +269,19 @@ sub _gotTrack {
 	my $cache = Slim::Utils::Cache->new;
 	$cache->set( 'wimp_meta_' . $info->{id}, $meta, 86400 );
 
-	if ($format =~ /mp4|aac|fla?c/i) {
-		my $http = Slim::Networking::Async::HTTP->new;
-		$http->send_request( {
-			request     => HTTP::Request->new( GET => $info->{url} ),
-			onStream    => $format =~ /fla?c/i ?
-						   \&Slim::Utils::Scanner::Remote::parseFlacHeader :
-						   \&Slim::Utils::Scanner::Remote::parseMp4Header,
-			onError     => sub {
-				my ($self, $error) = @_;
-				$log->warn( "could not find $format header $error" );
-				$params->{successCb}->();
-			},
-			passthrough => [ $song->track, { cb => sub {
-			                                    my $meta = $cache->get('wimp_meta_' . $info->{id});
-			                                    $meta->{bitrate} = sprintf("%.0f" . Slim::Utils::Strings::string('KBPS'), $song->track->bitrate/1000);
-			                                    $cache->set( 'wimp_meta_' . $info->{id}, $meta, 86400 );
-			                                    $params->{successCb}->();
-			                               } },
-			                 $info->{url} ],
+	Slim::Utils::Scanner::Remote::parseRemoteHeader( 
+		$song->track, $info->{url}, $format =~ s/flac/flc/r, 
```

Comment:
I should have statistics about Perl versions used. The `/r` modifier is 5.14+ only... and I didn't know about it! Thanks, learned something today. How did you figure it out?

Reply:
That said... would it make sense to have the replacement in `parseRemoteHeader` instead? Every caller will have to do this.

Reply:
I learnt that /r thing sometime ago, you now my "obsession" :smile: for (crazy) compact code, so I was looking for . But I did not know it was 5.14+, thanks. BTW, what is the oldest supported version? Is this 5.14? My test system is Windows so it's 5.14.
Re the replacement of flac/flc, yes I was wondering but to some extent the logic everywhere is the 3 letters for codecs , so it feels like giving a bad incentive to be flexible, I don't know. But one can make the case that mimetype is audio/flac so I'll change it if you prefer. 
BTW, on the stream.flac subject, I've used 'flac' because the convention, AFAIK, is different for file extension.

Reply:
Well, finally I did the change :smile:

Reply:
One thing about such code - it's purely for plugins, so what do you think about putting it in older version (like 8.x). I'm wondering because any change for plugin is almost useless if the plugin cannot use such shortcuts unless it relies on the very latest version of LMS.

Reply:
> Well, you're right - I did the change 😄

Which solved the backwards compatibility issue, too 😉.

Would you mind looking into the Deezer implementation, too, to see whether it could take advantage of this? Yes, changes which are only useful to plugin authors sometimes are hardly worth it. I myself probably have too many backward compatibility checks in place. And that's not really worth it for a code change which would only simplify things.

Reply:
Done for Deezer